sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -313,12 +328,48 @@ public RelDataType inferReturnType(SqlOperatorBinding sqlOperatorBinding)
     }
   }
 
+  private static class TimeColIdentifer extends SqlIdentifier
+  {
+
+    public TimeColIdentifer()
+    {
+      super("__time", SqlParserPos.ZERO);
+    }
+
+    @Override
+    public <R> R accept(SqlVisitor<R> visitor)
+    {
+
+      try {
+        return super.accept(visitor);
+      }
+      catch (CalciteContextException e) {
+        if (e.getCause() instanceof SqlValidatorException) {
+          throw DruidException.forPersona(DruidException.Persona.ADMIN)
+                              .ofCategory(DruidException.Category.INVALID_INPUT)
+                              .build(
+                                  e,
+                                  "Query could not be planned. A possible reason is [%s]",
```

Comment:
please add a test case to validate the error message.

Reply:
Didn't add that since they are already there, such as [testTimeColumnAggregationsOnLookups](https://github.com/gargvishesh/druid/blob/6e23061d48e98ec6fe53fb00a2dfd7c9dddf08c9/sql/src/test/java/org/apache/druid/sql/calcite/CalciteJoinQueryTest.java#L1491) and [testTimeColumnAggregationFromExtern](https://github.com/gargvishesh/druid/blob/cb050282a03fecf5d5af5678cfac6f7a73af7a43/extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQSelectTest.java#L1772)